### DISCUSSION TITLE
Confused about bundling .browser.js files

### DISCUSSION THREAD

**Discussion by santi-mir** [webpack/webpack#16855] [Q&A] [answered]
I compiled a project using tsc, and have some `filename1.js`, `filename2.js` and so on files that are incompatible with the browser (node js files), so I created the respective `filename1.browser.js` etc files, and added the mapping to the `package.json`.

My project looks like this
```shell
package.json
src/
lib-esm/
  filename.browser.js
  filename.js
  ...
demo/
  webpack.config.js
  src/index.html 
```

When I ask to bundle the files specifically telling `filename.browser.js` it wont really work because 
1. this file will point to `filename2.js` but webpack does not look into the `browser` key in the package.json
2. So it is not aware that importing `filename2.js` should be replaced by `filename2.browser.js`

How is this supposed to be achieved ? I am quite frustrated.

My Config

```javascript
const path = require('path');

const HtmlWebpackPlugin = require('html-webpack-plugin');

module.exports = {
target:"web",
  mode: 'development',
  entry: { "main":"./lib-esm/index.js" },
  output: {
    path: path.resolve(__dirname),
    filename: '[name].bundle.js',
  },
  plugins: [
    new HtmlWebpackPlugin({ template: './src/index.html' }),
  ],
  resolve:{
    aliasFields: ["browser"],
  }
};

```

**Comment by alexander-akait** [comment, marked as answer]
Please provide package.json with the `browser` field, you don't need `aliasFields: ["browser"],`

**Reply by santi-mir** [reply]
I found the error. The `browser` key indeed had a wrong path (actually, I thought that was how to specify), the correct path had a subfolder
```
browser: {
 "./lib-esm/subfolder/filename2.js": "..."
```

Thank you for replying.

**Comment by santi-mir** [comment]
@alexander-akait yes I am aware, because it will search by default according to the docs. But I was testing it.

```json
{
  "main": "./lib/index.js",
  "module": "./lib-esm/index.js",
  "types": "./lib/index.d.ts",
  "files": [
    "src",
    "lib",
    "lib-esm"
  ],
  "browser": {
    "./lib-esm/filename1.js": "./lib-esm/filename1.browser.js",
    "./lib-esm/filename2.js": "./lib-esm/filename2.browser.js"
  }
}
```

**Reply by alexander-akait** [reply]
So the problem has been solved, right? :smile:

**Reply by santi-mir** [reply]
Yes, I mark now
